runtime/src/main/kotlin/kotlin/native/concurrent/Worker.kt

```diff
@@ -34,14 +34,32 @@ public inline class Worker @PublishedApi internal constructor(val id: Int) {
          * Typically new worker may be needed for computations offload to another core, for IO it may be
          * better to use non-blocking IO combined with more lightweight coroutines.
          *
+         * @see init to initialize worker context on the current thread
          * @param errorReporting controls if an uncaught exceptions in the worker will be printed out
          */
         public fun start(errorReporting: Boolean = true): Worker = Worker(startInternal(errorReporting))
 
         /**
-         * Return the current worker, if known, null otherwise. null value will be returned in the main thread
-         * or platform thread without an associated worker, non-null - if called inside worker started with
-         * [Worker.start].
+         * Get or init worker event queue of the current thread, if not yet inited. This operation allows to use
+         * a work queue to any thread, not just thread created to host a worker with [Worker.start].
+         */
+        public val currentOrInit: Worker get() {
+            val id = initInternal(true)
+            return if (id != 0) Worker(id) else throw Error("Cannot init as worker")
+        }
+
+        /**
+         * Deinit worker event queue on the current thread. If called in worker context will lead to termination
+         * of the worker.
+         */
+        public fun currentDeinit() {
```

Comment:
Must be called automatically on thread death.

Reply:
Seriously reworked whole worker support and simplified API.